pkg/crc/errors/multierror.go

```diff
@@ -60,17 +60,18 @@ func (r RetriableError) Error() string {
 	return "Temporary error: " + r.Err.Error()
 }
 
-// RetryAfter retries a number of attempts, after a delay
-func RetryAfter(attempts int, callback func() error, d time.Duration) error {
+// RetryAfter retries for a certain duration, after a delay
+func RetryAfter(limit time.Duration, callback func() error, d time.Duration) error {
 	m := MultiError{}
-	for i := 0; i < attempts; i++ {
-		if i > 0 {
-			logging.Debugf("retry loop: attempt %d out of %d", i, attempts)
-		}
+	timeLimit := time.Now().Add(limit)
+	attempt := 0
+	for time.Now().Before(timeLimit) {
```

Comment:
This is a very theoritical concern, but would it be worth having `for time.Now().Before(timeLimit) && attempt < 2` so that we are guaranteed at least 2 tries, first one takes ages and times out, second one succeeds as soon as we try it because durin the "takes ages" part to of the first try, everything got setup as needed?

Reply:
Yes, it seems reasonable. I'm adding it.